client/components/tinymce/plugins/wpcom-view/wpvideo-view.jsx

```diff
@@ -0,0 +1,140 @@
+		const defaultAspectRatio = 16 / 9;
+		let { width, height } = this.props.video ? this.props.video : { width: defaultWidth, height: defaultWidth / defaultAspectRatio };
+		const aspectRatio = width / height;
+
+		if ( shortcodeWidthAttribute && ! shortcodeHeightAttribute ) {
+			width = shortcodeWidthAttribute;
+			height = shortcodeWidthAttribute / aspectRatio;
+		} else if ( ! shortcodeWidthAttribute && shortcodeHeightAttribute ) {
+			width = shortcodeHeightAttribute * aspectRatio;
+			height = shortcodeHeightAttribute;
+		} else if ( shortcodeWidthAttribute && shortcodeHeightAttribute ) {
+			const definedAspectRatio = shortcodeWidthAttribute / shortcodeHeightAttribute;
+			if ( definedAspectRatio > aspectRatio ) {
+				width = shortcodeHeightAttribute * aspectRatio;
+				height = shortcodeHeightAttribute;
+			} else {
+				width = shortcodeWidthAttribute;
+				height = shortcodeWidthAttribute / aspectRatio;
+			}
+		}
+
+		return { width, height };
+	}
+
+	getVideoAttributes() {
+		const shortcode = shortcodeUtils.parse( this.props.content );
+		const namedAttrs = shortcode.attrs.named;
+
+		const videoDimensions = this.constrainVideoDimensions( parseInt( namedAttrs.w, 10 ), parseInt( namedAttrs.h, 10 ) );
+
+		const shortcodeAttributes = {
+			guid: shortcode.attrs.numeric[0],
+			w: videoDimensions.width,
+			h: videoDimensions.height,
+			autoplay: namedAttrs.autoplay === 'true',
+			hd: namedAttrs.hd === 'true',
+			loop: namedAttrs.loop === 'true',
+			at: parseInt( namedAttrs.at, 10 ) || 0,
+			defaultLangCode: namedAttrs.defaultlangcode
+		};
+
+		return Object.assign( {}, pick( shortcodeAttributes, ['guid', 'w', 'h'] ), { embedUrl: this.getEmbedUrl( shortcodeAttributes ) } );
+	}
+
+	getEmbedUrl( shortcodeAttributes ) {
+		const defaultAttributeValues = { hd: false, at: 0, defaultLangCode: undefined };
+		const attributesWithNonDefaultValues = omit( shortcodeAttributes, ( value, key ) => defaultAttributeValues[key] === value );
+		const queryString = QueryString.stringify( pick( attributesWithNonDefaultValues, ['autoplay', 'hd', 'loop', 'at', 'defaultLangCode'] ) );
+
+		return `https://videopress.com/embed/${ shortcodeAttributes.guid }?${ queryString }`;
+	}
+
+	onLoad() {
+		const doc = tinymce.activeEditor.iframeElement.contentDocument;
+		const script = doc.createElement( 'script' );
+		script.src = 'https://videopress.com/videopress-iframe.js';
+		script.type = 'text/javascript';
+		doc.getElementsByTagName( 'head' )[0].appendChild( script );
+	}
+
+	render() {
+		const videoAttributes = this.getVideoAttributes();
+
+		return (
+			<div className="wpview-content">
+				<QueryVideo guid={ videoAttributes.guid } />
+				<iframe
+					onLoad={ this.onLoad }
+					width={ videoAttributes.w }
+					height={ videoAttributes.h }
+					src={ videoAttributes.embedUrl }
+					className="wpview-type-video"
+					frameBorder="0"
+					allowFullScreen />
+			</div>
+		);
+	}
+
+}
+
+WpVideoView.propTypes = {
+	siteId: PropTypes.number,
```

Comment:
You're not using this, so you ought remove it from `propTypes`. Define only those props the component itself needs, not what may or may not be passed.